### ISSUE TITLE
rcm, rvm, and Fish won't play nice

### ISSUE THREAD

**Issue by ezuk** [thoughtbot/rcm#138]
Hi there,

I use Fish as my shell, and had RVM working fine under Fish. Since installing rcm and putting my fish config files under RCM, RVM basically stopped detecting `cd` commands under Fish.

The fish config files are loaded (`rcup` worked, the symlinks are there, and all aliases are working). I'm not sure what the problem is, or even what information I can provide that would help. It broke right when I started using rcm. Any ideas?

Thanks, 
Erez

**Comment by mike-burns** [comment]
Interesting.

Do you have `rvm_project_rvmrc` set in `~/.rvmrc`?

Do you have a `~/.config/fish/functions/rvm.fish`? Is it readable? Does it contain the `__handle_rvmrc_stuff` function definition?

In general, what does your Fish/RVM setup look like?

**Comment by ezuk** [comment]
@mike-burns - thanks for taking a look.

I don't actually have a `~/.rvmrc` file -- oops. I don't think I ever had one, tbh. What should be in it?

my `~/.config/fish/functions/rvm.fish` is readable and this is what it says:

```
➜  ~ cat ~/.config/fish/functions/rvm.fish
function rvm --description='Ruby enVironment Manager'
  # run RVM and capture the resulting environment
  set --local env_file (mktemp -t rvm.fish.XXXXXXXXXX)
  bash -c 'source ~/.rvm/scripts/rvm; rvm "$@"; status=$?; env > "$0"; exit $status' $env_file $argv

  # apply rvm_* and *PATH variables from the captured environment
  and eval (grep '^rvm\|^[^=]*PATH\|^GEM_HOME' $env_file | grep -v '_clr=' | sed '/^[^=]*PATH/s/:/" "/g; s/^/set -xg /; s/=/ "/; s/$/" ;/; s/(//; s/)//')
  # clean up
  rm -f $env_file
end

function __handle_rvmrc_stuff --on-variable PWD
  # Source a .rvmrc file in a directory after changing to it, if it exists.
  # To disable this fature, set rvm_project_rvmrc=0 in $HOME/.rvmrc
  if test "$rvm_project_rvmrc" != 0
    set -l cwd $PWD
    while true
      if contains $cwd "" $HOME "/"
        if test "$rvm_project_rvmrc_default" = 1
          rvm default 1>/dev/null 2>&1
        end
        break
      else
        if test -e .rvmrc -o -e .ruby-version -o -e .ruby-gemset
          eval "rvm reload" > /dev/null
          eval "rvm rvmrc load" >/dev/null
          break
        else
          set cwd (dirname "$cwd")
        end
      end
    end

    set -e cwd
  end
end
```

So is the problem in the lack of `~/.rvmrc`? 

One of my projects does have a `.rvmrc` file, and when I `cd` into it, I get this:

```
➜  ~ cd mailcatcher/
You are using '.rvmrc', it requires trusting, it is slower and it is not compatible with other ruby managers,
you can switch to '.ruby-version' using 'rvm rvmrc to [.]ruby-version'
or ignore this warning with 'rvm rvmrc warning ignore /home/ezuk/mailcatcher/.rvmrc',
'.rvmrc' will continue to be the default project file in RVM 1 and RVM 2,
to ignore the warning for all files run 'rvm rvmrc warning ignore all.rvmrcs'.
```

That `.rvmrc` looks like this:

```
#!/usr/bin/env bash

# This is an RVM Project .rvmrc file, used to automatically load the ruby
# development environment upon cd'ing into the directory

# First we specify our desired <ruby>[@<gemset>], the @gemset name is optional,
# Only full ruby name is supported here, for short names use:
#     echo "rvm use 2.0.0" > .rvmrc
environment_id="ruby-2.0.0-p195@mailcatcher"

# Uncomment the following lines if you want to verify rvm version per project
# rvmrc_rvm_version="1.25.27 (master)" # 1.10.1 seems like a safe start
# eval "$(echo ${rvm_version}.${rvmrc_rvm_version} | __rvm_awk -F. '{print "[[ "$1*65536+$2*256+$3" -ge "$4*65536+$5*256+$6" ]]"}' )" || {
#   echo "This .rvmrc file requires at least RVM ${rvmrc_rvm_version}, aborting loading."
#   return 1
# }

# First we attempt to load the desired environment directly from the environment
# file. This is very fast and efficient compared to running through the entire
# CLI and selector. If you want feedback on which environment was used then
# insert the word 'use' after --create as this triggers verbose mode.
if [[ -d "${rvm_path:-$HOME/.rvm}/environments"
  && -s "${rvm_path:-$HOME/.rvm}/environments/$environment_id" ]]
then
  \. "${rvm_path:-$HOME/.rvm}/environments/$environment_id"
  for __hook in "${rvm_path:-$HOME/.rvm}/hooks/after_use"*
  do
    if [[ -f "${__hook}" && -x "${__hook}" && -s "${__hook}" ]]
    then \. "${__hook}" || true
    fi
  done
  unset __hook
  if (( ${rvm_use_flag:=1} >= 1 )) # display automatically
  then
    if [[ $- == *i* ]] # check for interactive shells
    then printf "%b" "Using: $(tput setaf 2 2>/dev/null)$GEM_HOME$(tput sgr0 2>/dev/null)
" # show the user the ruby and gemset they are using in green
    else printf "%b" "Using: $GEM_HOME
" # don't use colors in non-interactive shells
    fi
  fi
else
  # If the environment file has not yet been created, use the RVM CLI to select.
  rvm --create use  "$environment_id" || {
    echo "Failed to create RVM environment '${environment_id}'."
    return 1
  }
fi
```

**Comment by ezuk** [comment]
Problem solved! `rvm --create --versions-conf use 2.1.1@mygems`, then added the newly created `.versions.conf` to my `.gitignore`.

Thank you for setting me on the right path!
